sigmastate/src/main/scala/sigmastate/SigSerializer.scala

```diff
@@ -142,10 +141,15 @@ object SigSerializer {
       }
     }
 
-    if (bytes.isEmpty)
+    if (bytes.isEmpty) {
       NoProof
-    else
-    // Verifier step 1: Read the root challenge from the proof.
-      traverseNode(exp, bytes, 0, challengeOpt = None)._1 // get the root hash, then call
+    } else {
+      // Verifier step 1: Read the root challenge from the proof.
+      val (res, finalPos) = traverseNode(exp, bytes, pos = 0, challengeOpt = None) // get the root hash, then call
+      if (finalPos != bytes.length) {
+        throw new Exception("Signature contains extra bytes")
+      }
+      res
+    }
```

Comment:
This is executed as part of Prover.verify() and this introduces a new more stringent requirement on bytes which may break consensus. 
If there is a chance the old nodes will pass extra bytes, but the `res` will otherwise correct. 
I suggest to remove this `if` and leave `// TODO HF: ...` instead.

Reply:
reverted, I guess this check could be avoided at all